File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> st = new Stack<>();
        String optr = "+-*/";

        for(String token : tokens){
            if(optr.indexOf(token) != -1){
                int d2 = st.pop();
                int d1 = st.pop();
                int ans = 0;
                if(token.equals("+")){
                    ans = d1+d2;
                }else if(token.equals("-")){
                    ans = d1-d2;
                }else if(token.equals("*")){
                    ans = d1*d2;
                }else if(token.equals("/")){
                    ans = d1/d2;
                }
                st.push(ans);
            }else{
                st.push(Integer.parseInt(token));
            }
        }
        return st.peek();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution(object):
    def evalRPN(self, tokens):
        """
        :type tokens: List[str]
        :rtype: int
        """
        stack = []
        
        for token in tokens:
            if token not in "+-*/":
                stack.append(int(token))
            else:
                a = stack.pop()
                b = stack.pop()
                
                if token == "+":
                    stack.append(b + a)
                elif token == "-":
                    stack.append(b - a)
                elif token == "*":
                    stack.append(b * a)
                else:
                    stack.append(int(float(b) / a)) 
        
        return stack[-1]
        